lnbits/extension_manager.py

```diff
@@ -389,21 +400,26 @@ def download_archive(self):
 
     def extract_archive(self):
         logger.info(f"Extracting extension {self.name} ({self.installed_version}).")
-        Path("lnbits", "upgrades").mkdir(parents=True, exist_ok=True)
-        shutil.rmtree(self.ext_upgrade_dir, True)
+        Path(settings.lnbits_extensions_path, "upgrades").mkdir(
+            parents=True, exist_ok=True
+        )
+
+        tmp_dir = Path(settings.lnbits_data_folder, "unzip-temp", self.hash)
+        shutil.rmtree(tmp_dir, True)
+
         with zipfile.ZipFile(self.zip_path, "r") as zip_ref:
-            zip_ref.extractall(self.ext_upgrade_dir)
-        generated_dir_name = os.listdir(self.ext_upgrade_dir)[0]
-        os.rename(
-            Path(self.ext_upgrade_dir, generated_dir_name),
-            Path(self.ext_upgrade_dir, self.id),
+            zip_ref.extractall(tmp_dir)
+        generated_dir_name = os.listdir(tmp_dir)[0]
+        shutil.rmtree(self.ext_upgrade_dir, True)
+        shutil.copytree(
+            Path(tmp_dir, generated_dir_name),
+            Path(self.ext_upgrade_dir),
         )
+        shutil.rmtree(tmp_dir, True)
 
         # Pre-packed extensions can be upgraded
         # Mark the extension as installed so we know it is not the pre-packed version
-        with open(
-            Path(self.ext_upgrade_dir, self.id, "config.json"), "r+"
-        ) as json_file:
+        with open(Path(self.ext_upgrade_dir, "config.json"), "r+") as json_file:
```

Comment:
Should you check for existence of this file before opening?

Reply:
- if the file does not exist then `open` will throw
- this is fine `extract_archive` should fail in this case